File: placementprepration/celebprob.java
import java.util.*;
public class celebprob{
    public static boolean knows(int [][] matrix, int a, int b){
        return matrix[a][b]==1;
    }
    public static int findceleb(int [][] matrix,int n){
        Stack<Integer>st=new Stack<>();
        for(int i=0;i<n;i++){
            st.push(i);
        }
        while(st.size()>1){
            int a=st.pop();
            int b=st.pop();
            if(knows(matrix,a,b)){
                st.push(b);
            }
            else{
                st.push(a);
            }
        }
        int cand=st.pop();
        for(int i=0;i<n;i++){
            if(i!=cand){
                if(knows(matrix,cand,i) || !knows(matrix,i,cand)){
                    return -1;
                }
            }
        }
        return cand;

    }
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int [][]matrix=new int[n][n];
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                matrix[i][j]=sc.nextInt();
            }
        }
        int celeb=findceleb(matrix, n);
        if(celeb==-1){
            System.out.println("No celebrirt");
        }
        else{
            System.out.print(celeb);
        }

    }
}

File: placementprepration/celebprob.py
import sys


def knows(matrix, a, b):
    return matrix[a][b] == 1


def find_celeb(matrix):
    n = len(matrix)
    stack = list(range(n))
    while len(stack) > 1:
        a = stack.pop()
        b = stack.pop()
        if knows(matrix, a, b):
            stack.append(b)
        else:
            stack.append(a)
    cand = stack.pop()
    for i in range(n):
        if i == cand:
            continue
        if knows(cand_matrix := matrix, cand, i) or not knows(cand_matrix, i, cand):
            return -1
    return cand


def main():
    data = list(map(int, sys.stdin.read().strip().split()))
    if not data:
        return
    n, *rest = data
    vals = rest[: n * n]
    matrix = [vals[i * n : (i + 1) * n] for i in range(n)]
    celeb = find_celeb(matrix)
    if celeb == -1:
        print("No celebrirt")
    else:
        print(celeb)


if __name__ == "__main__":
    main()
